code/game/machinery/scanner_gate.dm

```diff
@@ -83,6 +105,17 @@
 	if(duration)
 		scanline_timer = addtimer(CALLBACK(src, PROC_REF(set_scanline), "passive"), duration, TIMER_STOPPABLE)
 
+/obj/machinery/scanner_gate/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = ..()
+	if(istype(tool, /obj/item/inspector))
```

Comment:
This needs to `return NONE` if it's not `istype(inspector)`, and should either return `ITEM_INTERACT_BLOCKING` or `ITEM_INTERACT_SUCCESS` depending on if one was inserted or not